---
title: "Discovering regulatory switches in complex biological networks"
author: "Ram√≥n Reszat, Luca Farinola, Patrick Mertens, Alaric Deboilly"
date: "2023-01-11"
output: html_document
---

```{r libraries, include=FALSE}
library(ggplot2) # create plots for the presentation
library(pcaMethods) # Bioconductor PCA methods
library(limma) # differential gene expression analysis (DGEA)
library(biomaRt) # download gene annotations from Ensembl
```

## Load MAGnet and Ensembl Datasets
First, load the original [MAGnet dataset](https://www.med.upenn.edu/magnet) and information on the study participants.
```{r MAGnet, warning=FALSE}
setwd('Data')

# transcriptome of each sample from the NCBI MAGnet dataset
gxData <- read.delim('MAGNET_GeneExpressionData_CPM_19112020.txt', header=TRUE, sep='\t', row.names=1)

# participant information for the tissue samples in the dataset
sampleInfo <- read.csv('MAGNET_SampleData_18112022.csv',header=TRUE, stringsAsFactor=TRUE)

# load the list of genes that are suspected biomarkers for type 2 diabetes
geneListT2D <- read.delim('MAGNET_Biomarkers_11012023.txt', header=FALSE, sep='\t')
```

Additionally, download gene annotations from the [Ensembl](http://www.ensembl.org/index.html) database using the biomaRt API.
```{r}
# connect to the Ensembl genes database and select homo sapiens as a species
ensembl = useEnsembl(biomart="genes", dataset="hsapiens_gene_ensembl", mirror='useast')

# download the Ensembl IDs for the genes selected from literature research
biomarkers <- getBM(attributes=c('ensembl_gene_id', 'hgnc_symbol'), filters='hgnc_symbol', values=geneListT2D, mart=ensembl)
```

## Differential Gene Expression Analysis

```{r design}
# if the baseline level is the healthy group limma will create the design matrix correctly
sampleInfo$etiology <- factor(sampleInfo$etiology, levels = c("NF", "DCM", "HCM", "PPCM"))

# creates dummy matrix and correct for confounding factors
design <- model.matrix(~0+etiology + gender + age, data=sampleInfo)
```

```{r contrasts}
# create contrasts for the linear model from the design matrix
contrasts <- makeContrasts(contrasts=c("etiologyDCM-etiologyNF"), levels=colnames(design))
```

$$
\text{Gene Expression} = 
\begin{cases}
\beta_0 + \beta_1 & \text{if case is DCM} \\
\beta_0 & \text{if case is NF}
\end{cases} + 
\begin{cases}
\beta_2 + \beta_3 & \text{if person is male} \\
\beta_2 & \text{if person is female} 
\end{cases} + 
\beta_4 \text{ Age}
$$
```{r model}
# fit linear regression for each gene
model <- lmFit(gxData, design)

# compute coefficients for each case-control pair
model  <- contrasts.fit(model, contrasts)

# Benjamini-Hochberg correction (FDR)
model <- eBayes(model, trend=TRUE)

head(model$coefficients)
```

```{r table}
# generate top-tables of p-values and log-fold changes
results <- topTable(model, coef=1, n=Inf, adjust="fdr")

head(results)
```